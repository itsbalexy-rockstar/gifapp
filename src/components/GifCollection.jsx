import GifElement from './GifElement';
import styled from "styled-components";
import { useFetchGifs } from '../hooks/useFetchGifs';

function GifCollection({category}) {
    
    const { data } = useFetchGifs(category)

  return (
    <GifCollectionStyled>
        <h1>{category}</h1>
        <div className='collection animate__animated animate__bounceInDown'>
        {
                data.map((gif) => (
                    <GifElement 
                    key={gif.id}
                    {...gif}
                    />
                ))
        }
        </div>
    </GifCollectionStyled>
    );
}

const GifCollectionStyled = styled.div`
    margin: 1rem;
    text-align: center;
    .collection{
        margin-top: 1rem;
        display: grid;
        grid-template-columns: auto auto auto;
        justify-content: center;
        gap: 1rem;
        padding: .5rem;
    }
    h1{
        text-transform: uppercase;
        background-color: #041562;
        padding: .5rem;
        border-radius: 20px;
        color: #eeeeee;
        font-size: 1.5rem;
    }
`

export default GifCollection;
