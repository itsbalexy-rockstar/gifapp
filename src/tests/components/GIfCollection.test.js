import React from 'react';
import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
import GifCollection from '../../components/GifCollection';
jest.mock('../../hooks/useFetchGifs')

describe('Test in <GifCollection />', () => {
    const category = 'One Punch'
    test('Should show the correctly component', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifCollection category={category} />)
        expect(wrapper).toMatchSnapshot()
    });
    test('Should show items when useFetch loaded the data', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything.jpg',
            title: 'Anything'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifCollection category={category} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifElement').length).toBe(gifs.length)
        
    });
    
})