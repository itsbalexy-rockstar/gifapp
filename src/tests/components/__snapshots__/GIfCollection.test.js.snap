// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifCollection /> Should show items when useFetch loaded the data 1`] = `
<styled.div>
  <h1>
    One Punch
  </h1>
  <div
    className="collection animate__animated animate__bounceInDown"
  >
    <GifElement
      id="ABC"
      key="ABC"
      title="Anything"
      url="https://localhost/anything.jpg"
    />
  </div>
</styled.div>
`;

exports[`Test in <GifCollection /> Should show the correctly component 1`] = `
<styled.div>
  <h1>
    One Punch
  </h1>
  <div
    className="collection animate__animated animate__bounceInDown"
  />
</styled.div>
`;
